src/posix/tty.rs

```diff
@@ -82,6 +83,35 @@ impl TTYPort {
         Ok(port)
     }
 
+    pub fn list_ports() -> Vec<SerialPortInfo> {
+        let context = libudev::Context::new().unwrap();
+        let mut enumerator = libudev::Enumerator::new(&context).unwrap();
+        enumerator.match_subsystem("tty").unwrap();
+        let mut vec = Vec::new();
+        for device in enumerator.scan_devices().unwrap() {
+            // Skip devices that don't have a parent. It's unclear why this works,
+            // but libserialport does it in their enumeration and it seems to work well.
```

Comment:
I don't think this comment is going to help anyone reading the code after this PR.